### PR TITLE
Unit test cases.

### PR DESCRIPTION
**Pull request by cnu1439**

### CODE REVIEW

File: blockonomics.js [hunk 1]
```diff
@@ -62,8 +98,14 @@ function sendRequest(fullpath, method, querystring) {
 				.end(function(response) {
 					resolve(response.body);
 				});
-
+		} else if (method == "DELETE") {
```
Comment (blockonomics) [inline blockonomics.js]: Remove this not needed

File: blockonomics.js [hunk 2]
```diff
@@ -31,10 +31,45 @@ module.exports = {
...
+		var url = config.config.newAddress;
+		if (reset == 1)
+			url = url + "reset=1";
+
+		if (match_account)
+			url = ( url + ( ( reset == 1 ) ? "&": "?" ) ) + "match_account=" + match_account;
+
+		var result = sendRequest(url, "POST", {}, APIKEY);
+		return result;
 	}
```
Comment (blockonomics) [inline blockonomics.js]: Are all methods returning errors properly ? Try calling newAddress.. without an xpub in account

Comment (cnu1439) [reply blockonomics.js]: yes.

Comment (blockonomics) [reply blockonomics.js]: Make API methods raise  exception with error message in case status is not 200

File: test/test.js [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    server
+    .get("/api/new_address/"+test_api_key)
+    .expect("Content-type",/json/)
+    .expect(200) // THis is HTTP response
+    .end(function(err,res){
+      // HTTP status should be 200
+      res.status.should.equal(200);
+      console.log(res.body);
+
+      if ( 'status' in res.body ) {
```
Comment (blockonomics) [inline test/test.js]: We never have 'status' in Json Body ?

File: test/test.js [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+
+describe("New Address", function(){
+  it("Should create new address", function(done){
+    server
+    .get("/api/new_address/"+test_api_key)
+    .expect("Content-type",/json/)
+    .expect(200) // THis is HTTP response
+    .end(function(err,res){
+      // HTTP status should be 200
+      res.status.should.equal(200);
```
Comment (blockonomics) [inline test/test.js]: compare the returned address

File: blockonomics.js (hunk 2 above)
Comment (cnu1439) [reply blockonomics.js]: ok

File: test/test.js (hunk 3 above)
Comment (cnu1439) [reply test/test.js]: In case of an error we are putting 'status' in the JSON response. which will tell the http response. 200 or 400 something like this.

Comment (blockonomics) [reply test/test.js]: this is same as res.status and not required to check... 

basic logic is if http status is not 200, then response will have error message

**Review by Godstimendako — COMMENTED** [review]
lpoqrt61ug9rorpz53mqb0ovdxh48mbtp9a1f49nz285azb5zxoluqj97ip02zmm

**Pull request merged**